src/download-cid.js

```diff
@@ -18,8 +18,11 @@ const SHORTCUT = IS_MAC
   : 'CommandOrControl+Alt+D'
 
 async function saveFile (dir, file) {
-  const location = join(dir, file.path)
-  await fs.outputFile(location, file.content)
+  const destination = join(dir, file.path)
+  if (!destination.startsWith(dir)) {
+    throw new Error(`unable to create '${file.path}' outside of '${dir}'`)
+  }
```

Comment:
Without knowing anything about anything in JS land: are UnixFS symlinks also under a similar check?

Reply:
Right now, symlinks are ignored by the download function in ipfs-desktop, only files and directories are written to disk.

But even if re-created symlinks, same destination check would be applied (symlink could point at external path, but its own location would be limited to destination dir).

Reply:
Just to be sure this won't become an issue in the future, added c4cfa325dec7b7e6862c5c1258f01e600e81457b